fbw-common/src/wasm/systems/systems/src/enhanced_gpwc/mod.rs

```diff
@@ -0,0 +1,184 @@
+            heading: Arinc429Word::new(Angle::new::<degree>(0.0), SignStatus::FailureWarning),
+            vertical_speed: Arinc429Word::new(
+                Velocity::new::<foot_per_minute>(0.0),
+                SignStatus::FailureWarning,
+            ),
+            navigation_display_range_lookup: range_lookup,
+            navigation_displays: [
+                NavigationDisplay::new(context, "L"),
+                NavigationDisplay::new(context, "R"),
+            ],
+            gear_is_down: true,
+            terronnd_rendering_mode,
+            egpwc_destination_longitude_id: context.get_identifier("EGPWC_DEST_LAT".to_owned()),
+            egpwc_destination_latitude_id: context.get_identifier("EGPWC_DEST_LONG".to_owned()),
+            egpwc_present_latitude_id: context.get_identifier("EGPWC_PRESENT_LAT".to_owned()),
+            egpwc_present_longitude_id: context.get_identifier("EGPWC_PRESENT_LONG".to_owned()),
+            egpwc_gear_is_down_id: context.get_identifier("EGPWC_GEAR_IS_DOWN".to_owned()),
+            egpwc_terronnd_rendering_mode: context
+                .get_identifier("EGPWC_TERRONND_RENDERING_MODE".to_owned()),
+        }
+    }
+
+    fn update_position_data(&mut self, adirs_output: &impl AdirsMeasurementOutputs) {
+        /*
+         * documentation hints:
+         *   - EGPWC has direct connection to GPS sensor && ADIRS_1
+         *   - uses direct GPS data if ADIRS_1 is unavailable
+         * TODO:
+         *   - implement logic as soon as GPS sensor is available
+         */
+        self.latitude = adirs_output.latitude(1);
+        self.longitude = adirs_output.longitude(1);
+        self.altitude = adirs_output.altitude(1);
+        self.heading = adirs_output.heading(1);
+        self.vertical_speed = adirs_output.vertical_speed(1);
+    }
+
+    pub fn update(
+        &mut self,
+        adirs_output: &impl AdirsMeasurementOutputs,
+        lgcius: &LandingGearControlInterfaceUnitSet,
+    ) {
+        self.update_position_data(adirs_output);
+        self.gear_is_down = lgcius.lgciu1().main_down_and_locked();
+
+        self.navigation_displays.iter_mut().for_each(|display| {
+            display.update(
+                &self.navigation_display_range_lookup,
+                adirs_output.is_fully_aligned(1),
+            )
+        });
+    }
+}
+
+impl SimulationElement for EnhancedGPWC {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_powered = buses.is_powered(self.powered_by)
+    }
+
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        let destination_long: f64 = reader.read(&self.fm1_destination_longitude_id);
+        let destination_lat: f64 = reader.read(&self.fm1_destination_latitude_id);
+        let destination_long_ssm: u32 = reader.read(&self.fm1_destination_longitude_ssm_id);
+        let destination_lat_ssm: u32 = reader.read(&self.fm1_destination_latitude_ssm_id);
```

Comment:
Because the current A429 binary format was designed without consideration for the Coherent/JS side, and it is not possible to write it from there. When writing a float value to a simvar it is cast to a single precision float along the way => corrupts the A429 value. The workaround we have can only be used for integer values, so writing an arbitrary double precision float doesn't work. https://github.com/flybywiresim/a32nx/pull/7766 addresses this but the rust x86_64 and wasm codegen appear to be a little different and the tests fail even though the aircraft runs fine in the sim. I'll push some fixes to that PR in the next few days.